ls-ecaps.c

```diff
@@ -950,17 +950,33 @@ dvsec_cxl_gpf_port(struct device *d, int where)
 }
 
 static void
-dvsec_cxl_flex_bus(struct device *d, int where, int rev)
+dvsec_cxl_flex_bus(struct device *d, int where, int rev, int len)
 {
   u16 w;
   u32 l, data;
 
-  if (rev < 1)
-  {
-    printf("\t\tRevision %d not supported\n", rev);
-    return;
+  // Sanity check: Does the length correspond to its revision?
+  switch (rev) {
+    case 0:
+      if (len != 0x10) {
```

Comment:
Length is the offset of the first unsupported register (first register after the last supported) in specified version, right? So what about using named register constant instead of magic number? E.g.

```suggestion
      if (len != PCI_CXL_FB_MOD_TS_DATA) {
```